pkg/logs/decoder/line_buffer.go

```diff
@@ -49,34 +45,25 @@ func (l *LineBuffer) AddEndOfLine() {
 }
 
 // AddIncompleteLine stores a chunck of line in buff
-func (l *LineBuffer) AddIncompleteLine(line *Line) {
-	l.buffer.Write(line.content)
-	l.contentLen += len(line.content)
+func (l *LineBuffer) AddIncompleteLine(line []byte) {
+	l.buffer.Write(line)
+	l.rawDataLen += len(line)
 }
 
 // AddTruncate stores TRUNCATED in buffer
-func (l *LineBuffer) AddTruncate(line *Line) {
+func (l *LineBuffer) AddTruncate(line []byte) {
 	l.buffer.Write(TRUNCATED)
 }
 
-// Flush creates a new output from content in buffer and sends it to outputChan
-func (l *LineBuffer) Flush() {
-	defer l.reset()
+// Content returns the content in buffer and the length of the data that enabled to compute this content
+func (l *LineBuffer) Content() ([]byte, int) {
 	content := make([]byte, l.buffer.Len())
 	copy(content, l.buffer.Bytes())
-	if len(content) > 0 {
-		output := NewOutput(content, l.contentLen)
-		l.outputChan <- output
-	}
+	return content, l.rawDataLen
 }
 
-// Stop forwards stop event to outputChan
-func (l *LineBuffer) Stop() {
-	l.outputChan <- newStopOutput()
-}
-
-// reset prepares buffer to receive new lines
-func (l *LineBuffer) reset() {
-	l.contentLen = 0
+// Flush prepares buffer to receive new lines
+func (l *LineBuffer) Flush() {
```

Comment:
could we keep the name `reset`? When I read flush for a buffer I think it's moving the content to the next stage, whereas here it's just reseting it